cloudinfo/catalog.go

```diff
@@ -443,9 +446,12 @@ func (infoSvc *CloudInfoService) processComponentReferencesWithGetter(addonConfi
 
 				found = true
 
-				// Process dependencies of this dependency recursively
-				if err := infoSvc.processComponentReferencesWithGetter(&addonConfig.Dependencies[i], processedLocators, getter); err != nil {
-					return err
+				// OPTIMIZATION: Only process dependencies of enabled optional dependencies
+				// This is safe because disabled optional dependencies won't be deployed anyway
+				if addonConfig.Dependencies[i].Enabled != nil && *addonConfig.Dependencies[i].Enabled {
```

Comment:
I don't think this will error or is incorrect, since you are just checking for nil, but the first reference is on the pointer itself, and the second condition is the actual value in the pointer.